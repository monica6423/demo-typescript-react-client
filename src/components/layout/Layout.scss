@import "../common.scss";

.container {
  width: 912px; 
  margin: 2rem auto;
  background-color: #f1f1f1;
}

.page {
  background-color: #f1f1f1;
}

.title {
  color: $main;
  margin-top: 64px;
  margin-bottom: 32px;
}

.header {
  background-color: #acb5bd;
  height: 96px;
}

.logo {
  display: flex;
  height: 100%;
  align-items: center;
  color: white;
  margin-left: 1rem;
}