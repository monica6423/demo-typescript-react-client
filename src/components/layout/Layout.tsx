import { ReactElement } from "react";
import { Link } from "react-router-dom";
import "./Layout.scss";

export interface LayoutProps {
  children?: ReactElement[] | ReactElement;
  title: string;
  link: string;
  linkTitle: string;
}

const Layout = ({ children, title, link, linkTitle }: LayoutProps) => {
  return (
    <div className="page">
      <div>
        <div className="header">
          <div className="logo">
            <h1>{title}</h1>
            <h3 style={{ marginLeft: "auto", marginRight: "1rem" }}>
              <Link to={link}>Go to {linkTitle} page âžœ </Link>
            </h3>
          </div>
        </div>
        <div className="container">{children}</div>
      </div>
    </div>
  );
};

export default Layout;
