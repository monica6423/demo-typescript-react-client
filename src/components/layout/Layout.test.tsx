import React from "react";
import { render, cleanup } from "@testing-library/react";
import Layout from "./Layout";
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

it("should render the component", () => {
  const { asFragment } = render(
    <Router>
      <Layout title="Test Title" link="/test" linkTitle="Test" />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should render the correct title", () => {
  const { getByText } = render(
    <Router>
      <Layout title="Test Title" link="/test" linkTitle="Test" />
    </Router>
  );
  expect(getByText("Test Title")).toBeInTheDocument();
});

it("should render the correct link", () => {
  const { getByText } = render(
    <Router>
      <Layout title="Test Title" link="/test" linkTitle="Test" />
    </Router>
  );
  const link = getByText(/Visit Test âžœ/i);
  expect(link.getAttribute("href")).toBe("/test");
});

it("should render the children", () => {
  const { getByText } = render(
    <Router>
      <Layout title="Test Title" link="/test" linkTitle="Test">
        <p>test</p>
      </Layout>
    </Router>
  );
  expect(getByText("test")).toBeInTheDocument();
});
