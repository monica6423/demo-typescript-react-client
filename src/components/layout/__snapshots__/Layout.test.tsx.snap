// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div>
      <div
        class="header"
      >
        <div
          class="logo"
        >
          <h1>
            Test Title
          </h1>
          <h3
            style="margin-left: auto; margin-right: 1rem;"
          >
            <a
              href="/test"
            >
              Go to Test page âžœ 
            </a>
          </h3>
        </div>
      </div>
      <div
        class="container"
      />
    </div>
  </div>
</DocumentFragment>
`;
